packages/webapp/src/utils/user.tsx

```diff
@@ -30,6 +29,14 @@ export function useSignout() {
         await logoutAPI(); // Destroy server session.
 
         await mutate(() => true, undefined, { revalidate: false }); // clean all cache
-        nav('/signin', { replace: true });
+
+        // swr/infinite doesn't currently support clearing cache keys with the
+        // default mechanism. see https://github.com/vercel/swr/issues/2497
+        for (const key of cache.keys()) {
+            await mutate(key, undefined, { revalidate: false });
+        }
+
+        // force a full reload to ensure all state is cleared
+        window.location.href = '/signin';
```

Comment:
I don't love this, but it's kind of a sledgehammer solution for cache busting as a fallback, and I doubt we have so many manual logouts that it really matters much.

Reply:
I think it's fair tbh
if we do that we can get rid of the cache busting anyway

Reply:
I figured I would leave the cache busting in there in case we ever decided we wanted a more permanent storage for the cache, but yeah, it's effectively just a few lines of useless code now.